### DISCUSSION TITLE
Should IPushDelegate.OnReceived be called on Android when the app is backgrounded?

### DISCUSSION THREAD

**Discussion by jmichas** [shinyorg/shiny#705] [Q&A] [answered]
I have looked through the documents and sample and can't seem to figure out if this should be getting called in the background or not. For me, OnReceived does not get called while my app is in the background. I am not sure if there is a configuration that needs to be done because the Android specifics area of the docs is TBD.

Thanks in advance for any info.

**Comment by aritchie** [comment, marked as answer]
Yes - you need to include a data payload.  For situations like this, you are best to refer to the OS documentation: https://firebase.google.com/docs/cloud-messaging/android/receive#backgrounded

**Reply by jmichas** [reply]
Ok, so using the Azure default android test:
````
{
	"notification":{
		"title":"Notification Hub Test Notification",
		"body":"This is a sample notification delivered by Azure Notification Hubs."
	},
	"data":{
		"property1":"value1",
		"property2":42
	}
}
````

It should fire in the background right?
I get a notification banner popup but the OnReceived doesn't fire.
I am also still on 2.1.0.2769, is this an issue with the older version?

Thanks.

**Reply by aritchie** [reply]
It will unless the service is killed for one reason or another.  I've been using it actively in projects since 2.1 without issue.

**Reply by Codelisk** [reply]
@jmichas i had it in the newest preview nuget (2.3.0.2855‑preview) that it did not work when i used services.UseFirebase, but it worked when i used service.UsePush instead. You can give that a try.

**Reply by aritchie** [reply]
@Codelisk I'm sorry, but this is not good advice.  First off, UsePush is moving in 2.3. Second, it is in preview and not complete.  Lastly, On Android - UseFirebase/UsePush are the exact thing on Android and UseFirebaseMessaging only applies if you're using Shiny.Push.FirebaseMessaging

**Reply by jmichas** [reply]
Ok, I have updated my code to use DryIoc and Shiny.Framework. Most of my issues have disappeared except this one.
I do get this in my debug log when I send a test notification when the app is in the background.
`[FirebaseMessaging] Missing Default Notification Channel metadata in AndroidManifest. Default value will be used.`

What should the default channel be? And could this be causing OnReceived to not fire?

**Reply by jmichas** [reply]
So, here is some progress on this. If I remove the `"notification":` section of the message and JUST send `data` it calls OnReceived.
Like this:
````
{
	"data":{
		"property1":"value1",
		"property2":42
	}
}
````
